Framework/Algorithms/src/CorrectToFile.cpp

```diff
@@ -55,7 +57,11 @@ void CorrectToFile::exec() {
   // Only create the output workspace if it's not the same as the input one
   MatrixWorkspace_sptr outputWS = getProperty("OutputWorkspace");
   if (outputWS != toCorrect) {
-    outputWS = WorkspaceFactory::Instance().create(toCorrect);
+    if (toCorrect->id() == "EventWorkspace") {
+      outputWS = create<Workspace2D>(*toCorrect);
```

Comment:
I am not sure why this extra handling for "EventWorkspace" is required. Does it not work if you use `create<HistoWorkspace>` in all cases?

Reply:
Apparently not, its something i carried over from the way tom was handling it

Reply:
I might have spoken too soon, I only checked with unit-tests, but it appears one of the changes made is causing failure in system tests.